x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -182,18 +187,19 @@ public void createResponse(String docId,
                                R response,
                                ActionListener<IndexResponse> listener) throws IOException {
         try {
-            // TODO: Integrate with circuit breaker
-            final XContentBuilder source = XContentFactory.jsonBuilder()
+            final ReleasableBytesStreamOutput buffer = new ReleasableBytesStreamOutput(0, bigArrays.withCircuitBreaking());
+            listener = ActionListener.runBefore(listener, buffer::close);
```

Comment:
Does this close need to be async, or could we just close immediately after retrieving the bytes?

Reply:
Yes, it needs to be async. We should not adjust the circuit breaker until the IndexRequest is no longer used.